airflow/api_connexion/schemas/dag_schema.py

```diff
@@ -0,0 +1,93 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import List, NamedTuple
+
+from marshmallow import Schema, fields
+from marshmallow_sqlalchemy import SQLAlchemySchema, auto_field
+
+from airflow.api_connexion.schemas.common_schema import TimeDeltaSchema, TimezoneField
+from airflow.models.dag import DagModel, DagTag
+
+
+class DagTagSchema(SQLAlchemySchema):
+    """Dag Tag schema"""
+    class Meta:
+        """Meta"""
+
+        model = DagTag
+
+    name = auto_field()
+
+
+class DAGSchema(SQLAlchemySchema):
+    """DAG schema"""
+
+    class Meta:
+        """Meta"""
+
+        model = DagModel
+
+    dag_id = auto_field(dump_only=True)
+    root_dag_id = auto_field(dump_only=True)
+    is_paused = auto_field(dump_only=True)
+    is_subdag = auto_field(dump_only=True)
+    fileloc = auto_field(dump_only=True)
+    owners = fields.Method("get_owners", dump_only=True)
+    description = auto_field(dump_only=True)
+    # schedule_interval = fields.Nested(ScheduleIntervalSchema, dump_only=True)
```

Comment:
Had to figure what's wrong:
```
FAILED [100%][2020-06-19 16:27:55,965] {validation.py:230} ERROR - http://localhost/api/v1/dags/test_dag/details validation error: {'__type': 'TimeDelta', 'days': 1, 'seconds': 0} is valid under each of {'x-scope': ['', '#/components/schemas/DAGCollection', '#/components/schemas/DAG', '#/components/schemas/ScheduleInterval'], 'type': 'object', 'required': ['__type'], 'properties': {'__type': {'type': 'string'}, 'years': {'type': 'integer'}, 'months': {'type': 'integer'}, 'days': {'type': 'integer'}, 'leapdays': {'type': 'integer'}, 'hours': {'type': 'integer'}, 'minutes': {'type': 'integer'}, 'seconds': {'type': 'integer'}, 'microseconds': {'type': 'integer'}, 'year': {'type': 'integer'}, 'month': {'type': 'integer'}, 'day': {'type': 'integer'}, 'hour': {'type': 'integer'}, 'minute': {'type': 'integer'}, 'second': {'type': 'integer'}, 'microsecond': {'type': 'integer'}}}, {'x-scope': ['', '#/components/schemas/DAGCollection', '#/components/schemas/DAG', '#/components/schemas/ScheduleInterval'], 'type': 'object', 'required': ['__type'], 'properties': {'__type': {'type': 'string'}, 'value': {'type': 'string'}}}, {'x-scope': ['', '#/components/schemas/DAGCollection', '#/components/schemas/DAG', '#/components/schemas/ScheduleInterval'], 'type': 'object', 'required': ['__type'], 'properties': {'__type': {'type': 'string'}, 'days': {'type': 'integer'}, 'seconds': {'type': 'integer'}, 'microsecond': {'type': 'integer'}}}
Failed validating 'oneOf' in schema['allOf'][0]['properties']['schedule_interval']:
    {'discriminator': {'propertyName': '__type'},
     'oneOf': [{'properties': {'__type': {'type': 'string'},
                               'days': {'type': 'integer'},
                               'microsecond': {'type': 'integer'},
                               'seconds': {'type': 'integer'}},
                'required': ['__type'],
                'type': 'object',
                'x-scope': ['',
                            '#/components/schemas/DAGCollection',
                            '#/components/schemas/DAG',
                            '#/components/schemas/ScheduleInterval']},
               {'properties': {'__type': {'type': 'string'},
                               'day': {'type': 'integer'},
                               'days': {'type': 'integer'},
                               'hour': {'type': 'integer'},
                               'hours': {'type': 'integer'},
                               'leapdays': {'type': 'integer'},
                               'microsecond': {'type': 'integer'},
                               'microseconds': {'type': 'integer'},
                               'minute': {'type': 'integer'},
                               'minutes': {'type': 'integer'},
                               'month': {'type': 'integer'},
                               'months': {'type': 'integer'},
                               'second': {'type': 'integer'},
                               'seconds': {'type': 'integer'},
                               'year': {'type': 'integer'},
                               'years': {'type': 'integer'}},
                'required': ['__type'],
                'type': 'object',
                'x-scope': ['',
                            '#/components/schemas/DAGCollection',
                            '#/components/schemas/DAG',
                            '#/components/schemas/ScheduleInterval']},
               {'properties': {'__type': {'type': 'string'},
                               'value': {'type': 'string'}},
                'required': ['__type'],
                'type': 'object',
                'x-scope': ['',
                            '#/components/schemas/DAGCollection',
                            '#/components/schemas/DAG',
                            '#/components/schemas/ScheduleInterval']}],
     'readOnly': True,
     'x-scope': ['',
                 '#/components/schemas/DAGCollection',
                 '#/components/schemas/DAG']}
On instance['schedule_interval']:
    {'__type': 'TimeDelta', 'days': 1, 'seconds': 0}
```

Reply:
The problem seems to be solved  by adding more than `__type` to `required` fields.